src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -528,6 +517,106 @@ private JsValue AppendTimeSeries(JsValue document, JsValue name, JsValue[] args)
+                        valuesArg = args[0];
+                        break;
+                    case 2:
+                        signature = signature2Args;
+                        timestamp = GetTimeSeriesDateArg(args[0], signature, "timestamp");
+                        valuesArg = args[1];
+                        break;
+                    default:
+                        throw new ArgumentException($"There is no overload with {args.Length} arguments for this method should be {signature1Args} or {signature2Args}");
+                }
+
+                var (id, doc) = GetIdAndDocFromArg(document, _timeSeriesSignature);
+
+                string timeSeries = GetStringArg(name, _timeSeriesSignature, "name");
+
+                double[] valuesBuffer = null;
+                try
+                {
+                    GetTimeSeriesValues(valuesArg, ref valuesBuffer, signature, out var values);
+
+                    var tss = _database.DocumentsStorage.TimeSeriesStorage;
+                    var newSeries = tss.Stats.GetStats(_docsCtx, id, timeSeries).Count == 0;
+
+                    if (newSeries)
+                    {
+                        DocumentTimeSeriesToUpdate ??= new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                        DocumentTimeSeriesToUpdate.Add(id);
+                    }
+
+                    var toIncrement = new TimeSeriesOperation.IncrementOperation
+                    {
+                        Values = values.ToArray(),
+                        Timestamp = timestamp
+                    };
+
+                    tss.IncrementTimestamp(
+                        _docsCtx,
+                        id,
+                        CollectionName.GetCollectionName(doc),
+                        timeSeries,
+                        new[] { toIncrement },
+                        AppendOptionsForScript);
+
+                    if (DebugMode)
+                    {
+                        DebugActions.IncrementTimeSeries.Add(new DynamicJsonValue
+                        {
+                            ["Name"] = timeSeries,
+                            ["Timestamp"] = timestamp,
+                            ["Values"] = values.ToArray().Cast<object>(),
+                            ["Created"] = newSeries
+                        });
+                    }
+                }
+                finally
+                {
+                    if (valuesBuffer != null)
+                        ArrayPool<double>.Shared.Return(valuesBuffer);
+                }
+
+                return Undefined.Instance;
+            }
+
+            private void GetTimeSeriesValues(JsValue valuesArg, ref double[] valuesBuffer, string signature, out Memory<double> values)
+            {
+                if (valuesArg.IsArray())
+                {
+                    var jsValues = valuesArg.AsArray();
+                    valuesBuffer = ArrayPool<double>.Shared.Rent((int)jsValues.Length);
```

Comment:
since you allocation later by `ToArray()` let's avoid this completely and handle the allocation optimization in a dedicated issue

Reply:
For now, I avoided allocation by adding a new property for `IncrementOperation` called `ValuesLength` (can be `null`), as we discussed, and sending `valuesBuffer` with the values' length.
Hope that's fine :-)